### ISSUE TITLE
Add padding from input when keyboards open

### ISSUE THREAD

**Issue by CarterBland** [FaridSafi/react-native-gifted-chat#1609]
#### Issue Description

When the keyboard is open on Android there's no space between the keyboard and the focused input. I haven't found any props that adds distance between the input and the keyboard.

#### Steps to Reproduce / Code Snippets

![Screen Shot 2020-01-15 at 8 34 07 AM](https://user-images.githubusercontent.com/16313465/72445742-d4db5000-3777-11ea-8c12-df384aa0d100.png)

#### Expected Results

Ability to add distance between input to account for the send button.

#### Additional Information

    System:
      OS: macOS 10.15.1
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 12.6.0 - /usr/local/bin/node
      Yarn: 1.13.0 - /usr/local/bin/yarn
      npm: 6.13.0 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.3 AI-182.5107.16.33.5199772
      Xcode: 11.3/11C29 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^35.0.0 => 35.0.1 
      react: 16.8.3 => 16.8.3 
      react-native: https://github.com/expo/react-native/archive/sdk-35.0.0.tar.gz => 0.59.8 
    npmGlobalPackages:
      expo-cli: 3.11.2

**Comment by shamilovtim** [comment]
Is this related to your problem? #1424

**Comment by CarterBland** [comment]
@shamilovtim Hey, that's not my issue, I don't run into that. My issue is that I can't find a way to add padding between the bottom of text input and the top of the keyboard to make space for my custom button. Everything else works besides that.

**Comment by shamilovtim** [comment]
@CarterBland the keyboard shouldn't be covering your input like this, that's why i asked. it's not as simple as needing to program a padding, the input rests on top of the keyboard, thus the keyboard does not cover the input by default. there is a different bug happening i think. do you have a header at the top of the chat?

**Comment by CarterBland** [comment]
@shamilovtim I have a custom send button, the keyboard isn't covering the input, it's actually working perfectly fine with the input, too good almost. It's stuck to the actual bottom of the input on Android, and I can't get it to move.

**Comment by shamilovtim** [comment]
Sorry but I don't understand. If the bottom of your input was attached to the top of your keyboard, you wouldn't need any kind of padding. In the screenshot you provided the keyboard is covering your input.

**Comment by CarterBland** [comment]
@shamilovtim Hi, the input itself isn't covered, nor is it inhibited by the keyboard in any sense. The keyboard looks like it's covering it, but the only thing it's actually covering is the custom send button, it is, though, close to the input but not covering it.

**Comment by DnEgorWeb** [comment]
I have exactly the same problem. My InputToolbar's height is a bit bigger than default value. When keyboard is active it covers my InputToolbars (it doesn't cover the whole input, just part of it, as on the screenshot of @CarterBland )

**Comment by shamilovtim** [comment]
@DnEgorWeb Yeah I was trying to explain this up above to @CarterBland. The keyboard by default should not cover any part of your input regardless of whether you have something inside of the input or have made the input larger. The input rests on top of the keyboard, it is the only thing on top of the keyboard. There's no padding to apply to anything because there's nothing aside from the input above the keyboard. Long story short this looks like a bug.

**Comment by shamilovtim** [comment]
Correct me if I'm wrong @DnEgorWeb but in the interrim you can immediately fix this issue by making your button fit within the constraints of the default TextInput size and remove any manual sizing from the TextInput?

**Comment by shamilovtim** [comment]
Feel free to post your code here

**Comment by CarterBland** [comment]
@shamilovtim Yeah, I believe it's a bug too, do you know of any fixes we could apply besides changing the size?

**Comment by shamilovtim** [comment]
@CarterBland It will be easier to debug if you can a) provide whole code of the chatscreen and app.jsx b) provide a .gif of the behavior including triggering the keyboard, and typing in the input.

**Comment by DnEgorWeb** [comment]
Sorry for late reply guys. I found out that I didn't change android:windowSoftInputMode to "adjustResize" for the chat. I needed `adjustPan` in my application, so I used `react-native-android-keyboard-adjust` library to change `android:windowSoftInputMode` to `adjustResize` when my chat is mounted and return it back when the component is unmounted. Once I changed the mode it started to work perfectly, all my paddings are presented when keyboard is open. So if it does the trick for @CarterBland I assume we can close the issue.
